torchao/prototype/autoround/autoround_demo.py

```diff
@@ -0,0 +1,95 @@
+import argparse
+
+import torchao
+
+from torchao.prototype.autoround.core import (
+    auto_round_config,
+    MultiTensor,
+    post_process_model_after_applying_auto_round_,
+    prepare_model_for_applying_auto_round_,
+)
+
+
+def main(args):
+    # 0. Get the model, tokenizer, and decoder_cls
+    import torchao.prototype.autoround.utils as ar_utils
+
+    model_name_or_path = args.model_name_or_path
+    model, tokenizer, decoder_cls = ar_utils.get_float_model_info(model_name_or_path)
+    # Workaround for disabling the `kv_cache`, which cause the OOM.
+    model.config.use_cache = False
+    ar_utils.gen_text(model, tokenizer, "Float model", device="cuda", max_length=50)
+
+    auto_round_config.iters = args.iters
+    auto_round_config.nsamples = args.nsamples
+    auto_round_config.seqlen = args.seqlen
+
+    # 1. Prepare the model for applying auto-round
+    # User should provide the `is_decoder` function for identifying the decoder block
+    # It can be extended to other modules, such as `lm_head`, the function like:
+    #   is_target_module = lambda mod, fqn: isinstance(mod, decoder_cls) or "lm_head" in fqn
+    if args.quant_lm_head:
+        # is_decoder = lambda mod, fqn:  "lm_head" in fqn
+        is_decoder = lambda mod, fqn: isinstance(mod, decoder_cls) or "lm_head" in fqn
+    else:
+        is_decoder = lambda mod, fqn: isinstance(mod, decoder_cls)
+    prepare_model_for_applying_auto_round_(model, is_decoder)
```

Comment:
@jerryzh168 here is the new flow using `MulTensor`, it's quite similar to the flow using `prepare` and `convert`, Does that sound okay to you?

Reply:
yeah using MultiTensor sounds fine, and the flow makes sense, see my suggestion on the implementation below